CONTRIBUTING.txt

```diff
@@ -170,6 +170,43 @@ Once you've fixed all merge conflicts, do::
    <https://scikit-image.org/docs/dev/gitwash/development_workflow.html#rebasing-on-trunk>`__,
    but we squash and merge most PRs either way.
 
+Build environment setup
+-----------------------
+
+Development for scikit-image is possible on Windows, Linux, and MacOSX.
+Once you've cloned your fork of the git repository as detailed above,
+you should setup a Python development environment tailored for scikit-image.
+
+pip+virtualenv
+--------------
+If you choose to use pip and virtualenv, then you may find these
+instructions helpful::
+
+  virtualenv env
+  source env/bin/activate
+  pip install -r requirements/default.txt
+  pip install -r requirements/build.txt
+  pip install -r requirements/test.txt
```

Comment:
How about the documentation? We insist that all new feature must have an example, so it's important to consider it as part of the build env. setup.

Reply:
My thoughts are that **new** devs don't necessary want to run all the examples. They will likely just install the deps they need for the examples they are developing. I'm trying to help people to push PR for the first time, not necessarily make the best example right off the bat.

We might also need:
1. Beginners contribution guide
2. Advanced contribution guide

Reply:
My thought is that some new devs start with a new feature as a first PR. It not negligible in scikit-image.

Reply:
Alright. Do you think it is appropriate to bundle it all in the main requirements.txt file? having 4 commands is a little annoying.

Reply:
Sadly not: I don't think we want `requirements.txt` to contain everything, since most people look there for "minimal requirements". I just found this workaround:

```
pip install -r <(cat requirements/*.txt)
```

https://github.com/pypa/pip/issues/3880

Be sure to add a comment to that line to the effect of, "install all development, runtime, and documentation requirements of scikit-image."

Reply:
Any workaround for conda? I don't want to maintain a yaml file...